app/forms/base_assets_form.rb

```diff
@@ -2,6 +2,9 @@ class BaseAssetsForm
   include ActiveModel::Model
   include ActiveModel::Attributes
   include SessionPersistable
+  include NumberValidatable
+
+  validate :numbers_are_all_valid
```

Comment:
You don't actually need to specify this in each Form file, because the `included` block in NumberValidatable means that this validation gets automatically declared by anything that includes it.